docs/source/programs_and_gates.rst

```diff
@@ -240,6 +242,63 @@ program should be executed 1000 times:
     The word “shot” comes from experimental physics where an experiment is
     performed many times, and each result is called a shot.
 
+.. _build_a_fixed_count_loop:
+
+Build a fixed-count loop with Quil
+----------------------------------
+
+Specifying trials with :py:meth:`~pyquil.quil.Program.wrap_in_numshots_loop` doesn't modify the Quil in your program in
+any way. Instead, the number of shots you specify is included in your job request and tells the executor how many times
+to run your program. However, with Quil's :ref:`classical_control_flow`, instructions it is possible to write a program
+that itself defines a loop over a number of shots. The :py:meth:`~pyquil.quil.Program.with_loop` method will help you
+do just that. It wraps the body of your program in a loop over a number of iterations you specify and returns the looped
+program.
+
+Let's see an example. We'll construct a classic bell state program and measure it 1000 times by wrapping the program in a
+Quil loop.
+
+.. testcode:: with_loop
+
+    from pyquil import Program, get_qc
+    from pyquil.quilatom import Label
+    from pyquil.gates import H, CNOT
+
+    # Setup the bell state program
+    p = Program(
+        H(0),
+        CNOT(0, 1),
+    )
+    ro = p.declare("ro", "BIT", 2)
+    p.measure(0, ro[0])
+    p.measure(1, ro[1])
+
+    # Declare a memory region to hold the number of shots
+    shot_count = p.declare("shot_count", "INTEGER")
+
+    # Wrap the program in a loop by specifying the number of iterations, a memory reference to
+    # hold the number of iterations, and two labels to mark the beginning and end of the loop.
+    looped_program = p.with_loop(1000, shot_count, Label("start-loop"), Label("end-loop"))
+    print(looped_program.out())
+
+    qc = get_qc("2q-qvm")
+    # Specify your desired shot count in the memory map.
+    results = qc.run(looped_program, memory_map={"shot_count": [1000]})
```

Comment:
Why specify `1000` here _and_ in `with_loop()`?

Reply:
Oops, yeah, got confused with the other examples I was working on and [this comment](https://github.com/rigetti/pyquil/pull/1717#discussion_r1431579399) 😅